bigchaindb/core.py

```diff
@@ -494,36 +486,20 @@ def validate_block(self, block):
         """
         return self.consensus.validate_block(self, block)
 
-    def has_previous_vote(self, block_id, voters):
+    def has_previous_vote(self, block_id):
         """Check for previous votes from this node
 
         Args:
             block_id (str): the id of the block to check
-            voters (list(str)): the voters of the block to check
 
         Returns:
             bool: :const:`True` if this block already has a
             valid vote from this node, :const:`False` otherwise.
 
-        Raises:
-            ImproperVoteError: If there is already a vote,
-                but the vote is invalid.
-
         """
         votes = list(backend.query.get_votes_by_block_id_and_voter(self.connection, block_id, self.me))
-
-        if len(votes) > 1:
-            raise exceptions.MultipleVotesError('Block {block_id} has {n_votes} votes from public key {me}'
-                                                .format(block_id=block_id, n_votes=str(len(votes)), me=self.me))
-        has_previous_vote = False
-        if votes:
-            if utils.verify_vote_signature(voters, votes[0]):
-                has_previous_vote = True
-            else:
-                raise exceptions.ImproperVoteError('Block {block_id} already has an incorrectly signed vote '
-                                                   'from public key {me}'.format(block_id=block_id, me=self.me))
-
-        return has_previous_vote
+        el, _ = self.consensus.voting.partition_eligible_votes(votes, [self.me])
```

Comment:
Would this happen in practice? Vote pipeline listens to block changefeed, so it shouldn't get itself into a loop. Also, should we even be calling has_previous_vote from the vote pipeline? is there a case where it could return True and so a vote is not cast and the pipeline is functioning correct with respect to `previous_block`?